proposals/text/7462-rag-evaluation.md

```diff
@@ -0,0 +1,243 @@
+        "context_relevance": [0.805466, 0.410251, 0.750070, 0.361332],
+        "faithfulness": [0.135581, 0.695974, 0.749861, 0.041999],
+        "semantic_answer_similarity": [0.971241, 0.159320, 0.019722, 1],
+        "aggregated_score":
+          {
+              'Reciprocal Rank': 0.448,
+              'Single Hit': 0.5,
+              'Multi Hit': 0.540,
+              'Context Relevance': 0.537,
+              'Faithfulness': 0.452,
+              'Semantic Answer Similarity': 0.478
+          }
+    },
+    "pipeline_2": {
+        "reciprocal_rank": [0.378064, 0.534964, 0.216058, 0.778642],
+        "single_hit": [1, 1, 0, 1],
+        "multi_hit": [0.706125, 0.454976, 0.445512, 0.250522],
+        "context_relevance": [0.805466, 0.410251, 0.750070, 0.361332],
+        "faithfulness": [0.135581, 0.695974, 0.749861, 0.041999],
+        "semantic_answer_similarity": [0.971241, 0.159320, 0.019722, 1],
+        "aggregated_score":
+          {
+              'Reciprocal Rank': 0.448,
+              'Single Hit': 0.5,
+              'Multi Hit': 0.540,
+              'Context Relevance': 0.537,
+              'Faithfulness': 0.452,
+              'Semantic Answer Similarity': 0.478
+          }
+      }
+}
+```
+
+Have a method to find interesting scores thresholds, typically used for error analysis, for all metrics available.
+Some potentially interesting thresholds to find are: the 25th percentile, the 75th percentile, the mean , the median.
+
+```python
+def find_thresholds(self, metrics: List[str]) -> Dict[str, float]:
+```
+
+```bash
+data  = {
+    "thresholds": ["25th percentile", "75th percentile", "mean", "average"],
+    "reciprocal_rank": [0.378064, 0.534964, 0.216058, 0.778642],
+    "context_relevance": [0.805466, 0.410251, 0.750070, 0.361332],
+    "faithfulness": [0.135581, 0.695974, 0.749861, 0.041999],
+    "semantic_answer_similarity": [0.971241, 0.159320, 0.019722, 1],
+}
+````
+
+Then have another method that  
+
+```python
+def get_scores_below_threshold(self, metric: str, threshold: float):
+    """Get the all the queries with a score below a certain threshold for a given metric"""  
+```
+
+
+# Drawbacks
+
+- Relying on pandas DataFrame internally makes it easy to perform many of the operations.
+- Nevertheless, it can be burden, since we are making `pandas` a dependency of `haystack-ai`.
+- Ideally all the proposed methods should be implemented in a way that doesn't require `pandas`.
+
+
+# Adoption strategy
+
+- Doesn't introduce any breaking change, it is a new feature that can be adopted by users as they see fit for their use cases.
+
+# How we teach this
+
+- A tutorial would be the best approach to teach users how to use this feature.
+- Adding a new entry to the documentation.
+
+# Unresolved questions
+
+- The `comparative_summary()` and the `comparative_detailed_summary()` methods need to be adopted to different definitions of what a correct answer is.
```

Comment:
Not sure I understand the semantics here? Comparisons should only provide a way to display the data side-by-side - We shouldn't be concerned with the semantics of the displayed metrics in any way.